dateparser/search/search.py

```diff
@@ -188,10 +195,11 @@ def detect_language(self, text, languages):
         else:
             self.language_detector = FullTextLanguageDetector(list(self.available_language_map.values()))
 
-        return self.language_detector._best_language(text)
+        detected_language = self.language_detector._best_language(text) or settings.DEFAULT_LANGUAGES[0]
+        return detected_language
 
     @apply_settings
-    def search_dates(self, text, languages=None, settings=None):
+    def search_dates(self, text, languages=None, detect_languages_func=None, settings=None):
```

Comment:
same about putting the `detect_languages_func` as the last parameter to avoid a breaking change.